src/main/java/uk/gov/justice/digital/hmpps/whereabouts/services/court/EventStoreListener.kt

```diff
@@ -0,0 +1,86 @@
+package uk.gov.justice.digital.hmpps.whereabouts.services.court
+
+import org.springframework.beans.factory.annotation.Qualifier
+import org.springframework.stereotype.Component
+import org.springframework.transaction.annotation.Transactional
+import uk.gov.justice.digital.hmpps.whereabouts.dto.VideoLinkBookingSpecification
+import uk.gov.justice.digital.hmpps.whereabouts.dto.VideoLinkBookingUpdateSpecification
+import uk.gov.justice.digital.hmpps.whereabouts.model.VideoLinkBooking
+import uk.gov.justice.digital.hmpps.whereabouts.model.VideoLinkBookingEvent
+import uk.gov.justice.digital.hmpps.whereabouts.model.VideoLinkBookingEventType
+import uk.gov.justice.digital.hmpps.whereabouts.repository.VideoLinkBookingEventRepository
+import uk.gov.justice.digital.hmpps.whereabouts.security.AuthenticationFacade
+import java.time.Clock
+import java.time.LocalDateTime
+
+@Component
+@Transactional
+@Qualifier("delegate")
+class EventStoreListener(
```

Comment:
I am not sure about using a listener for updating the event table. This should be done before we send the telemetry event, given that it is all done in a JPA transaction. The telemetry client has no concept of transactions, so it should always be the last thing that is done.

Reply:
hmm.  OK, I should be able to refactor this.  I thought it might be tricky because the application has 'open session in view' configured for JPA, but I just checked and I can't find it now.

Reply:
Done. 
Turns out open session in view is the default in Spring Boot so turned that off in the configuration.